pkg/gradle/gradle.go

```diff
@@ -74,3 +79,69 @@ func getParametersFromOptions(options *ExecuteOptions) []string {
 
 	return parameters
 }
+
+// CreateBOM generates BOM file using CycloneDX
+func createBOM(options *ExecuteOptions, utils Utils) error {
+	// check if gradle task cyclonedxBom exists
+	stdOutBuf := new(bytes.Buffer)
+	stdOut := log.Writer()
+	stdOut = io.MultiWriter(stdOut, stdOutBuf)
+	utils.Stdout(stdOut)
+	if err := utils.RunExecutable(exec, "tasks"); err != nil {
+		return fmt.Errorf("failed list gradle tasks: %w", err)
+	}
+	if strings.Contains(stdOutBuf.String(), bomTaskName) {
+		if err := utils.RunExecutable(exec, bomTaskName); err != nil {
+			return fmt.Errorf("BOM creation failed: %w", err)
+		}
+	} else {
+		if err := createInitScript(options.BuildGradlePath); err != nil {
+			return err
+		}
+		defer deleteInitScript(options.BuildGradlePath)
+		if err := utils.RunExecutable(exec, "--init-script", filepath.Join(options.BuildGradlePath, initScriptName), bomTaskName); err != nil {
+			return fmt.Errorf("BOM creation failed: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func createInitScript(buildGradlePath string) error {
+	initScript, err := os.Create(filepath.Join(buildGradlePath, initScriptName))
+	if err != nil {
+		return fmt.Errorf("failed create init script: %w", err)
+	}
+	defer initScript.Close()
+
+	initScriptText := `
+initscript {
+  repositories {
```

Comment:
repositories might not be enough, especially when using other repos which are potentially even protected.
Do we have a change to install the cycloneDX plugin using the team config file?

Reply:
So far, I find it impossible to pass the plugin and settings via flags to gradle (as we do for maven). We need to add the plugin through the configuration files. In our case, through an init script.